kernel/tty/src/channel.rs

```diff
@@ -0,0 +1,95 @@
+// TODO: Use more efficient data structure?
+// TODO: A sender and receiver are the same under the hood. Hypothetically we
+// could store just one.
+// FIXME: async_channel is not a proper mpmc
+// FIXME: Error handling
+
+use async_channel::{Receiver, Sender};
+use core2::io::{ErrorKind, Result};
+
+#[derive(Clone)]
+pub(crate) struct Channel {
```

Comment:
Not quite. This makes `read` blocking when operating on buffers. It will block on the first byte, and then use `try_read`/`try_write` to operate on the remainder of the bytes.

Reply:
oh right, i skimmed it too quickly. 

Could be useful on its own too, right? Perhaps consider moving it into a separate crate since it's essentially just a wrapper around a channel.

Reply:
Moved the methods to `async_channel` in 595948c. Now the module is just to convert the `async_channel::Error` to `core2::io::Error`.